### ISSUE TITLE
Dragging a selection over closed folds causes spurious errors and error popup

### ISSUE THREAD

**Issue by albertdev** [vrapper/vrapper#657]
Scenario:

* Take any file with folding (e.g. a Java class with multiple imports and a number of methods).
* Close all folds using `zM`.
* Drag a selection over several folds.
* Soon a popup like the following shows up:

![failed to get location info for v-1](https://cloud.githubusercontent.com/assets/1451362/8350577/1999dc4c-1b27-11e5-8da9-37632bc1d592.PNG)

In the error log the following stacktrace gets printed:

~~~
net.sourceforge.vrapper.platform.VrapperPlatformException: Failed to get location info for V-1
	at net.sourceforge.vrapper.eclipse.platform.EclipseCursorAndSelection.updateStickyColumn(EclipseCursorAndSelection.java:795)
	at net.sourceforge.vrapper.eclipse.platform.EclipseCursorAndSelection.access$2(EclipseCursorAndSelection.java:789)
	at net.sourceforge.vrapper.eclipse.platform.EclipseCursorAndSelection$SelectionChangeListener.widgetSelected(EclipseCursorAndSelection.java:430)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:248)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
~~~

This impacts users of 0.56.0 (and possibly earlier versions too).

---

The cause seems to be in the `getPosition()` function or the selection code in general: when calculating the "to" member of a `Selection`, it doesn't take folds into account and that might cause the "to" member to point to a model offset which is hidden in a fold.

When `updateStickyPosition()` runs, it will call `getPosition()` to know where the caret is at and calculate the view column. However, model offsets hidden in a fold have no corresponding view offsets and so the view offset conversion returns `-1`. The code in `updateStickyPosition()` doesn't check for this and crashes when passing `-1` to other Eclipse APIs.

Ignoring any requests when we receive a view offset of `-1` might sidestep the issue, but the sticky column might never get updated otherwise. The problem ultimately remains in the calculations for the `Selection`, and it should be fixed there.

**Comment by albertdev** [comment]
It looks like this got introduced in the 0.50.0 release.

Version 0.48.0 did not have this problem, though it did have some other bug where dragging a selection might sometimes be aborted.

**Comment by albertdev** [comment]
It looks like this might not be *too* hard to fix, I guess I have to change some of the calculations so that they will use view offsets.

**Comment by keforbes** [comment]
For anyone following this defect, the fix has been included in the latest unstable build (0.57.20150702).
